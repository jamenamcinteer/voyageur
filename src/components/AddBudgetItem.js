import React from "react";
import TripHeader from "./Navigation/TripBudgetHeader";
import BudgetItemForm from "./Forms/BudgetItemForm";
import { connect } from "react-redux";

const AddBudgetItem = props => {
  const trip = props.trips.find(trip => trip._id === props.match.params.id);

  const budgetCategory = props.budgetCategories.find(
    budgetCategory => budgetCategory._id === props.match.params.budgetCategoryId
  );

  return (
    <React.Fragment>
      {budgetCategory && (
        <React.Fragment>
          <TripHeader
            title="Add Budget Item"
            theme={props.theme}
            backTo={`/trip-budget/${props.match.params.id}`}
            trip={trip}
            budgetItems={props.budgetItems}
            expenses={props.expenses}
            auth={props.auth}
          />
          <BudgetItemForm
            theme={props.theme}
            trip={trip}
            budgetCategory={budgetCategory}
            history={props.history}
          />
        </React.Fragment>
      )}
    </React.Fragment>
  );
};

const mapStateToProps = (state, props) => {
  return {
    trips: state.trips,
    budgetCategories: state.budgetCategories,
    budgetItems: state.budgetItems,
    expenses: state.expenses,
    auth: state.auth
  };
};

export default connect(mapStateToProps)(AddBudgetItem);
