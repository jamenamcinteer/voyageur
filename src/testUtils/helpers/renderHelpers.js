import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import theme from "../../theme";
import "jest-styled-components";
import { createStore } from "redux";
import { Provider } from "react-redux";
import tripsReducer, { tripsReducerDefaultState } from "../../reducers/trips";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

export function renderWithTheme(component) {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
}

export function renderWithReduxRouterAndTheme(
  ui,
  {
    initialState,
    store = createStore(tripsReducer, tripsReducerDefaultState),
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Router history={history}>{ui}</Router>
        </ThemeProvider>
      </Provider>
    ),
    store,
    history
  };
}
