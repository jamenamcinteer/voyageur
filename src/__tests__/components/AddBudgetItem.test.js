import React from "react";
import { cleanup } from "@testing-library/react";
import { createStore } from "redux";
import "react-dates/initialize";
import AddBudgetItem from "../../components/AddBudgetItem";
import {
  trips,
  budgetCategories,
  budgetItems,
  expenses,
  auth
} from "../../testUtils/fixtures/mockStoreData";
import { renderWithReduxRouterAndTheme } from "../../testUtils/helpers/renderHelpers";

afterEach(cleanup);

jest.mock("uniqid", () => {
  let value = 1;
  return () => value++;
});

const store = createStore(() => ({
  trips,
  budgetCategories,
  budgetItems,
  expenses,
  auth
}));

test("can render with redux with custom store", () => {
  const { asFragment } = renderWithReduxRouterAndTheme(
    <AddBudgetItem match={{ params: { id: "1", budgetCategoryId: "1" } }} />,
    {
      store
    }
  );
  expect(asFragment()).toMatchSnapshot();
});
