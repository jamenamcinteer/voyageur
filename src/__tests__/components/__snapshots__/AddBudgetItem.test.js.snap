// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render with redux with custom store 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <div>
          <div
            class="sc-bdVaJa hZqfbl"
          >
            <a
              href="/trip/1"
            >
              <svg
                style="fill: #4b5c6b; width: 20px; height: 20px; float: left; margin-top: 2px;"
                version="1.1"
                viewBox="0 0 199.404 199.404"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <polygon
                    points="135.412,0 35.709,99.702 135.412,199.404 163.695,171.119 92.277,99.702 163.695,28.285 	"
                  />
                </g>
              </svg>
            </a>
            <div
              class="sc-bwzfXH ebZROL"
            >
              Add Budget Item
            </div>
            <div
              class="sc-htpNat KtFsv"
            >
              <img
                alt=""
                class="sc-bxivhb bfhPyW"
                src="https://picsum.photos/id/947/45/45"
              />
            </div>
          </div>
          <div
            style="position: relative;"
          >
            <div
              style="background-image: url(https://images.unsplash.com/photo-1549317487-92cec5e76645?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjc5ODc0fQ&w=300&h=100&fit=crop&crop=focalpoint); background-repeat: no-repeat; background-size: cover; filter: brightness(0.6); height: 150px;"
            />
            <div
              style="height: 150px; padding: 0px 20px; color: rgb(255, 255, 255); font-size: 1em; font-weight: normal; position: absolute; width: 100%; top: 0px;"
            >
              <div
                style="display: flex; justify-content: space-between; align-items: center; height: 90px;"
              >
                <h2>
                  Hawaii
                </h2>
                <h3>
                  Jan 1, 1970 - Oct 1, 8307
                </h3>
              </div>
              <div
                style="width: 100%; height: 25px; position: relative; background: rgb(220, 224, 227); border-radius: 5px;"
              >
                <span
                  style="display: block; height: 100%; border-radius: 5px; position: relative; overflow: hidden; width: 29.731%; background-color: rgb(26, 174, 159);"
                />
                <div
                  style="position: absolute; top: 0px; color: white; left: 10px; line-height: 25px;"
                >
                  $595 of $2000
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-cSHVUG bhLOSl"
      >
        <h3
          class="sc-kpOJdX iUFbzx"
        >
          <span>
            Budget Category: Airfare
          </span>
          <a
            backgroundcolor="transparent"
            border="0"
            buttontype="link"
            class="sc-fjdhpX bbvnea"
            href="/trip/1"
            style="padding: 0px 10px; margin: 0px; width: auto; font-size: .8em;"
          >
            <span
              class="sc-jzJRlG cphvgw"
            >
              Change
            </span>
          </a>
        </h3>
        <div
          class="sc-bZQynM blxRiI"
        >
          <label
            class="sc-gzVnrw sohPS"
            for="1"
          >
            Name of Budget Item
          </label>
          <input
            class="sc-htoDjs hNiZih"
            id="1"
            type="text"
            value=""
          />
        </div>
        <div />
        <div
          class="sc-bZQynM blxRiI"
        >
          <label
            class="sc-gzVnrw sohPS"
            for="2"
          >
            Estimated Cost ($)
          </label>
          <input
            class="sc-htoDjs hNiZih"
            id="2"
            placeholder="0.00"
            type="text"
            value=""
          />
        </div>
        <div />
        <div
          class="sc-dnqmqq jnbCEU"
        >
          <label
            class="sc-iwsKbI dtSDPi"
            for="3"
          >
            Notes (optional)
          </label>
          <textarea
            class="sc-gZMcBi kEtTwg"
            id="3"
            type="text"
          />
        </div>
        <div
          class="sc-kgoBCf eZNNpy"
        >
          <div
            class="sc-chPdSV hATzHZ"
          >
            <a
              backgroundcolor="transparent"
              border="0"
              buttontype="link"
              buttonwidth="auto"
              class="sc-fjdhpX bbvnea"
              href="/trip/1"
              style="padding: 10px 0px;"
            >
              <span
                class="sc-jzJRlG cphvgw"
              >
                Cancel
              </span>
            </a>
            <button
              class="sc-VigVT kMgnqA"
            >
              <span
                class="sc-jTzLTM cntDpR"
                color="#fff"
              >
                Save
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
